File: CS/RichEditAPISample/CodeUtils/CodeResultViewer.cs
using System;
using System.CodeDom.Compiler;
using System.IO;
using System.Reflection;
using System.Windows.Forms;
using Microsoft.CSharp;
using System.Globalization;
using DevExpress.Spreadsheet;

namespace RichEditAPISample
{
    public abstract class RichEditExampleCodeEvaluator : ExampleCodeEvaluator
    {


        protected override string GetModuleAssembly()
        {
            return AssemblyInfo.SRAssemblyRichEditCore;
        }
        protected override string GetExampleClassName()
        {
            return "RichEditCodeResultViewer.ExampleItem";
        }
    }
    #region RichEditCSExampleCodeEvaluator
    public class RichEditCSExampleCodeEvaluator : RichEditExampleCodeEvaluator
    {

        protected override CodeDomProvider GetCodeDomProvider()
        {
            return new CSharpCodeProvider();
        }
        const string codeStart =
      "using System;\r\n" +
      "using DevExpress.XtraRichEdit;\r\n" +
      "using DevExpress.XtraRichEdit.API.Native;\r\n" +
      "using System.Drawing;\r\n" +
      "using System.Windows.Forms;\r\n" +
      "using DevExpress.Utils;\r\n" +
      "using System.IO;\r\n" +
      "using System.Diagnostics;\r\n" +
      "using System.Xml;\r\n" +
      "using System.Data;\r\n" +
      "using System.Collections.Generic;\r\n" +
      "using System.Linq;\r\n" +
      "using System.Globalization;\r\n" +
      "namespace RichEditCodeResultViewer { \r\n" +
      "public class ExampleItem { \r\n" +
      "        public static void Process(Document document) { \r\n" +
      "\r\n";

        const string codeEnd =
        "       \r\n }\r\n" +
        "    }\r\n" +
        "}\r\n";
        protected override string CodeStart { get { return codeStart; } }
        protected override string CodeEnd { get { return codeEnd; } }
    }
    #endregion
    #region RichEditVbExampleCodeEvaluator
    public class RichEditVbExampleCodeEvaluator : RichEditExampleCodeEvaluator
    {

        protected override CodeDomProvider GetCodeDomProvider()
        {
            return new Microsoft.VisualBasic.VBCodeProvider();
        }
        const string codeStart =
      "Imports Microsoft.VisualBasic\r\n" +
      "Imports System\r\n" +
      "Imports DevExpress.XtraRichEdit\r\n" +
      "Imports DevExpress.XtraRichEdit.API.Native\r\n" +
      "Imports System.Drawing\r\n" +
      "Imports System.Windows.Forms\r\n" +
      "Imports DevExpress.Utils\r\n" +
      "Imports System.IO\r\n" +
      "Imports System.Diagnostics\r\n" +
      "Imports System.Xml\r\n" +
      "Imports System.Data\r\n" +
      "Imports System.Core\r\n" +
      "Imports System.Collections.Generic\r\n" +
      "Imports System.Globalization\r\n" +
      "Namespace RichEditCodeResultViewer\r\n" +
      "	Public Class ExampleItem\r\n" +
      "		Public Shared Sub Process(ByVal document As Document)\r\n" +
      "\r\n";

        const string codeEnd =
        "\r\n		End Sub\r\n" +
        "	End Class\r\n" +
        "End Namespace\r\n";

        protected override string CodeStart { get { return codeStart; } }
        protected override string CodeEnd { get { return codeEnd; } }
    }
    #endregion
}

File: VB/RichEditAPISample/CodeUtils/CodeResultViewer.vb
Imports System.CodeDom.Compiler
Imports Microsoft.CSharp

Namespace RichEditAPISample

    Public MustInherit Class RichEditExampleCodeEvaluator
        Inherits ExampleCodeEvaluator

        Protected Overrides Function GetModuleAssembly() As String
            Return AssemblyInfo.SRAssemblyRichEditCore
        End Function

        Protected Overrides Function GetExampleClassName() As String
            Return "RichEditCodeResultViewer.ExampleItem"
        End Function
    End Class

'#Region "RichEditCSExampleCodeEvaluator"
    Public Class RichEditCSExampleCodeEvaluator
        Inherits RichEditExampleCodeEvaluator

        Protected Overrides Function GetCodeDomProvider() As CodeDomProvider
            Return New CSharpCodeProvider()
        End Function

        Const codeStartField As String = "using System;" & Microsoft.VisualBasic.Constants.vbCrLf & "using DevExpress.XtraRichEdit;" & Microsoft.VisualBasic.Constants.vbCrLf & "using DevExpress.XtraRichEdit.API.Native;" & Microsoft.VisualBasic.Constants.vbCrLf & "using System.Drawing;" & Microsoft.VisualBasic.Constants.vbCrLf & "using System.Windows.Forms;" & Microsoft.VisualBasic.Constants.vbCrLf & "using DevExpress.Utils;" & Microsoft.VisualBasic.Constants.vbCrLf & "using System.IO;" & Microsoft.VisualBasic.Constants.vbCrLf & "using System.Diagnostics;" & Microsoft.VisualBasic.Constants.vbCrLf & "using System.Xml;" & Microsoft.VisualBasic.Constants.vbCrLf & "using System.Data;" & Microsoft.VisualBasic.Constants.vbCrLf & "using System.Collections.Generic;" & Microsoft.VisualBasic.Constants.vbCrLf & "using System.Globalization;" & Microsoft.VisualBasic.Constants.vbCrLf & "namespace RichEditCodeResultViewer { " & Microsoft.VisualBasic.Constants.vbCrLf & "public class ExampleItem { " & Microsoft.VisualBasic.Constants.vbCrLf & "        public static void Process(Document doc) { " & Microsoft.VisualBasic.Constants.vbCrLf & Microsoft.VisualBasic.Constants.vbCrLf

        Const codeEndField As String = "       " & Microsoft.VisualBasic.Constants.vbCrLf & " }" & Microsoft.VisualBasic.Constants.vbCrLf & "    }" & Microsoft.VisualBasic.Constants.vbCrLf & "}" & Microsoft.VisualBasic.Constants.vbCrLf

        Protected Overrides ReadOnly Property CodeStart As String
            Get
                Return codeStartField
            End Get
        End Property

        Protected Overrides ReadOnly Property CodeEnd As String
            Get
                Return codeEndField
            End Get
        End Property
    End Class

'#End Region
'#Region "RichEditVbExampleCodeEvaluator"
    Public Class RichEditVbExampleCodeEvaluator
        Inherits RichEditExampleCodeEvaluator

        Protected Overrides Function GetCodeDomProvider() As CodeDomProvider
            Return New Microsoft.VisualBasic.VBCodeProvider()
        End Function

        Const codeStartField As String = "Imports Microsoft.VisualBasic" & Microsoft.VisualBasic.Constants.vbCrLf & "Imports System" & Microsoft.VisualBasic.Constants.vbCrLf & "using DevExpress.XtraRichEdit;" & Microsoft.VisualBasic.Constants.vbCrLf & "using DevExpress.XtraRichEdit.API.Native;" & Microsoft.VisualBasic.Constants.vbCrLf & "Imports System.Drawing" & Microsoft.VisualBasic.Constants.vbCrLf & "Imports System.Windows.Forms" & Microsoft.VisualBasic.Constants.vbCrLf & "Imports DevExpress.Utils" & Microsoft.VisualBasic.Constants.vbCrLf & "Imports System.IO" & Microsoft.VisualBasic.Constants.vbCrLf & "Imports System.Diagnostics" & Microsoft.VisualBasic.Constants.vbCrLf & "Imports System.Xml" & Microsoft.VisualBasic.Constants.vbCrLf & "Imports System.Data" & Microsoft.VisualBasic.Constants.vbCrLf & "Imports System.Collections.Generic" & Microsoft.VisualBasic.Constants.vbCrLf & "Imports System.Globalization" & Microsoft.VisualBasic.Constants.vbCrLf & "Namespace RichEditCodeResultViewer" & Microsoft.VisualBasic.Constants.vbCrLf & Microsoft.VisualBasic.Constants.vbTab & "Public Class ExampleItem" & Microsoft.VisualBasic.Constants.vbCrLf & Microsoft.VisualBasic.Constants.vbTab & Microsoft.VisualBasic.Constants.vbTab & "Public Shared Sub Process(ByVal doc As Document)" & Microsoft.VisualBasic.Constants.vbCrLf & Microsoft.VisualBasic.Constants.vbCrLf

        Const codeEndField As String = Microsoft.VisualBasic.Constants.vbCrLf & Microsoft.VisualBasic.Constants.vbTab & Microsoft.VisualBasic.Constants.vbTab & "End Sub" & Microsoft.VisualBasic.Constants.vbCrLf & Microsoft.VisualBasic.Constants.vbTab & "End Class" & Microsoft.VisualBasic.Constants.vbCrLf & "End Namespace" & Microsoft.VisualBasic.Constants.vbCrLf

        Protected Overrides ReadOnly Property CodeStart As String
            Get
                Return codeStartField
            End Get
        End Property

        Protected Overrides ReadOnly Property CodeEnd As String
            Get
                Return codeEndField
            End Get
        End Property
    End Class
'#End Region
End Namespace
